### PR TITLE
Basic Structure of service to filter by email

### PR DESCRIPTION
**Pull request by shivamvinsol**

### CODE REVIEW

File: app/services/spree/customer_segmentation/base_service.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
+module Spree
+  class CustomerSegmentation::BaseService
+
+    def initialize(collection)
+      @collection = collection
```
Comment (sawangupta92) [inline app/services/spree/customer_segmentation/base_service.rb]: att_accessor

File: app/services/spree/customer_segmentation/search_service.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+module Spree
+  class CustomerSegmentation::SearchService
+
+    # Assuming that args will give us formatted values
+    # Format values inside controller, call service
+    def initialize(args = {})
+      @collection = User.all
```
Comment (sawangupta92) [inline app/services/spree/customer_segmentation/search_service.rb]: attr_accessor

File: app/services/spree/customer_segmentation/search_service.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      @options = [ { term: 'user_email__includes', value: ['abc@mail.com', 'xyz@mail.com'] } ]  #args[:options]
+      # @options = [ { term: 'user_email__does_not_includes', value: ['abc@mail.com', 'xyz@mail.com'] } ]  #args[:options]
+    end
+
+    def perform
+      @options.each do |option|
+        service_name_key = get_search_key_name(option[:term])
+        operator = get_operator(option[:term])
+
+        @result = CUSTOMER_SEGMENTATION_SERVICE_MAPPER[service_name_key].new(@collection, operator, option[:value]).filter_data
```
Comment (sawangupta92) [inline app/services/spree/customer_segmentation/search_service.rb]: rename CUSTOMER_SEGMENTATION_SERVICE_MAPPER

**Review by sawangupta92 — CHANGES_REQUESTED** [review]

File: app/controllers/spree/admin/customer_segmentation_controller.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      # SERVICE ??
+      def create_values
+        if params[:operator] && params[:operator] =~ /include/
+          params[:value].split # convert to array
+        else
+          params[:value]
+        end
+      end
+
+      def apply_sorting
```
Comment (sawangupta92) [inline app/controllers/spree/admin/customer_segmentation_controller.rb]: why do we need this?

File: app/models/spree/user_decorator.rb [hunk 5]
```diff
@@ -0,0 +1,21 @@
+Spree::User.class_eval do
+
+  def name
```
Comment (sawangupta92) [inline app/models/spree/user_decorator.rb]: let's delegate them to address.

File: app/controllers/spree/admin/customer_segmentation_controller.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+      @results = @query.result.page(params[:page])
+    end
+
+    private
+
+      def process_params
+        { term: create_term, values: create_values }
+      end
+
+      def create_term
```
Comment (sawangupta92) [inline app/controllers/spree/admin/customer_segmentation_controller.rb]: let's use two different params instead metric and operator separately

**Review by sawangupta92 — CHANGES_REQUESTED** [review]

**Comment by shivamvinsol** [comment]
closing, work included in #3

**Pull request closed without merging**